### ISSUE TITLE
Multiple figures don't show

### ISSUE THREAD

**Issue by Kay-Da** [sillsdev/ptx2pdf#309]
I have a couple of places in my Paratext project where there are multiple figures in the same place, however, when I use PTXprint to make a PDF it only shows one (or none) of those figures.

Otherwise, PTXprint is much nicer than PrintDRAFT, so I'd love to be able to use it!

**Comment by Kay-Da** [comment]
I get an error if a heading (\s1 marker) comes after an \iex marker. 
It doesn't show as an error in Paratext and is how I want it to be. Is there any way I can get PTXprint to ignore it?

**Comment by markpenny** [comment]
@Kay-Da it would reallt be helpful if you can let us know which version you are running. There have been some improvements made in the pictures area for version 1.3.8 (which was released about 24 hours ago). If you aren't already using this latest version, then please update and then let us know if you are still facing the same issue.

IF the issue persists, then please use the button on the Help tab to create a ZIP archive (or a small subset of the data with the issue, like 1 small book) and then post a link here, or if it is a sensitive project, then e-mail the ZIP file to ptxprint_support@sil.org

This will help us identify the issue sooner, and hopefully fix it for you.

**Comment by Kay-Da** [comment]
Thanks, I've just updated to PTXprint 1.4 and the issue persists. I guess it's pretty rare to have multiple pictures in a row! We only have it a few times for comparisons. I wasn't sure how to choose a book to Zip/what I zipped. 1 Corinthians would be a good example (3:5-8). So let me know how to get the info you need from help\Create Zip Archive if you need something more.

![image](https://user-images.githubusercontent.com/74755226/100061978-688cab00-2e76-11eb-8406-8e879fc9ee3f.png)

![image](https://user-images.githubusercontent.com/74755226/100165438-cadebd80-2f01-11eb-8294-b77afdb9b5ca.png)


[ATV-DefaultPTXprintArchive.zip](https://github.com/sillsdev/ptx2pdf/files/5588097/ATV-DefaultPTXprintArchive.zip)

**Comment by mhosken** [comment]
Yes there is a bug in the code that parses usfm to extract figues in that it only matches the first \fig in any verse. You can have more than one figure per verse, but you'll need to add them in the ptxprint pictures interface.

**Comment by markpenny** [comment]
@mhosken has just enhanced the figure handling to make this possible. It will be in 1.4.1 (hopefully released in the next day or 2). Note that you will need to regenerate the PicList on the Pictures > Details tab. 
![image](https://user-images.githubusercontent.com/12962708/100214681-09fa2680-2f75-11eb-9245-97d7f69f5b1e.png)
(Notice the 3 figures listed for the same verse-range. And please ignore the error where the figures aren't listed in order! - we're still working on that one).

But yes - you can anchor multiple pictures to the same verse now - Yay!

![image](https://user-images.githubusercontent.com/12962708/100214629-fc44a100-2f74-11eb-8e13-4f43f5f32f6a.png)

**Comment by Kay-Da** [comment]
Thanks!

Is there any way to get alerted when an update is released? And is there an
easy way to update, or do I just download the program again?

**Comment by markpenny** [comment]
@Kay-Da - thanks for confirming that this has been fixed.

Unfortunately we don't have a was for PTXprint to auto-update itself (like Paratext does). But at least you now have a link on the Help tab to the PTXprint Home page (where you can always download the latest version and install it).

**Comment by Kay-Da** [comment]
Thanks Mark,the update link is useful. If there's an email list or anything
to get updates about updates, I'd be interested to get them.

I actually hadn't tested the multiple picture fix as 1.4.1 wasn't released
yet when I replied. I've now given it a go. It puts all three pictures on
the page, but not in the correct order (it put the first picture in the
series in the top right, the next in top left and the next bottom left). I
noticed when I regenerated the pictures that there was a tick box "Randomly
position pictures on page". When I unticked that box and regenerated the
pictures, it put my three pictures in order in the left hand column, but
the bottom one was  half below the line at the bottom and had the footnote
through it. Are there any fine tuning options I should try?

Thanks,

Kathy


Kathy Dadd

Australian Society for Indigenous Languages


0424 068 531

**Comment by markpenny** [comment]
Hi @Kay-Da  The best way to position the illustrations more precisely is to use the Vertical Posizition and Horizontal Position options on the Details sub-tab of the main Pictures tab as shown here:
![image](https://user-images.githubusercontent.com/12962708/101453942-03b47300-3995-11eb-8c54-cb2f4439b9fe.png)

You will notice that the furthest right column in the screen-grab above shows the short form for the position (tl=top left, bl=bottom left, etc.) of each illustration.

BTW - By unchecking the "randomly position..." box you have essentially told it to place all pictures at the top (left) corner and stack them one after the other - which it has duly done, right off the end of the page! So, I would suggest you regenerate (random) positions, and then adjust the ones that need changing.

**Comment by Kay-Da** [comment]
Hey Mark,

I had another go at using PTX. I updated it again as the version I had (I
think it was 1.4.6) didn't seem to have the details tab for pictures. I'm
now using 1.5.4 and get this error for a number of books:
[image: image.png]

After clicking ok on each, it gives this error and freezes:
[image: image.png]


Any idea what it relates to?

Thanks,

Kathy


Kathy Dadd

Australian Society for Indigenous Languages


0424 068 531

**Comment by mhosken** [comment]
Sorry Kay. The images didn't come through either on the email or in the issue could you go to https://github.com/sillsdev/ptx2pdf/issues/309 and edit the issue there with the images?

**Comment by Kay-Da** [comment]
First error (which shows for 7 books)
![image](https://user-images.githubusercontent.com/74755226/105105861-7ec48980-5afc-11eb-9689-4f89a9cf74c3.png)
Then is has: 
![image](https://user-images.githubusercontent.com/74755226/105105933-a3206600-5afc-11eb-84b9-2b4cfeaac641.png)
and freezes. 
It may be because a colleague added some pictures. I added them to my local figures folder, but I'm not sure what else I need to do. I tried re-generating the pictures, but it didn't help.

**Comment by markpenny** [comment]
@Kay-Da Could you please [download 1.5.8](https://software.sil.org/downloads/r/ptxprint/SetupPTXprint(1.5.8).exe) and see if this issue is resolved. If so, I'll close it - or you could do so yourself.

**Comment by Kay-Da** [comment]
Now I get this error (even if I just try one book):

![image](https://user-images.githubusercontent.com/74755226/106213789-043eec80-6215-11eb-8f39-f6201ecfda80.png)

**Comment by markpenny** [comment]
@Kay-Da Just checking whether this issue has been resolved in later versions. We are up to [PTXprint version 1.5.17](https://software.sil.org/downloads/r/ptxprint/SetupPTXprint(1.5.17).exe) now and are hoping to release 1.6 by the end of this week. It would be good to know if this issue can be closed. Thanks!

**Comment by Kay-Da** [comment]
I downloaded the latest version, but there are just other issues now. I'm happy for you close the issue anyway, as I don't think I'll be using PTX print for a while.

I'm using PTXprint 1.5.17

When I tried to 'print' everything, I got this error:
![image](https://user-images.githubusercontent.com/74755226/110587713-16bf2500-81c8-11eb-85a3-8c7dc179bd35.png)
Probably because we added blank lines after \iex lines, but we need them for typesetting.

When I did just 1 Cor it produced a PDF, but the formatting isn't great:
![image](https://user-images.githubusercontent.com/74755226/110587809-37877a80-81c8-11eb-9238-266bdee11891.png)
The headers and page numbers have the top margin line going through them.
And even though the pictures are set to "column width" they stay in the middle, unless I manually change them to "left", which is what I did below. I also made the below ones 80% so they'd fit. I'm not sure how to get the text to fill the previous page though.

![image](https://user-images.githubusercontent.com/74755226/110589216-31929900-81ca-11eb-95c6-5f09d369bb87.png)

I think what I was wanting might just be a bit too complicated for PTX print, especially with my skills.

Thanks anyway,

Kathy
